qdrant_client/qdrant_remote.py

```
@@ -270,38 +289,75 @@ def _parse_url(url: str) -> tuple[Optional[str], str, Optional[int], Optional[st
+        This method should be preferred over accessing `self._pool_size` directly as it applies the
+        default value if no pool_size was provided.
+        """
+
+        if self._pool_size is not None:
+            return self._pool_size
+        else:
+            return self.DEFAULT_GRPC_POOL_SIZE
+
     def _init_grpc_channel(self) -> None:
         if self._closed:
             raise RuntimeError("Client was closed. Please create a new QdrantClient instance.")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)